### ISSUE TITLE
Accounts will always get `FAIL_INVALID` after unstaking

### ISSUE THREAD

**Issue by pathornteng** [hashgraph/hedera-services#13854]
### Description

If an account stakes to a node and then unstakes by setting the staking node to 0.0.0, it will no longer be able to submit transactions successfully and will always get a `FAIL_INVALID` error for all transactions.

### Steps to reproduce

1. Stake an account to a node
```
  const transaction = await new AccountUpdateTransaction()
    .setAccountId(myAccountId)
    .setStakedAccountId("0.0.7")
    .execute(client);
```
2. Unstake the account by setting node ID to `0.0.0`
```
  const transaction = await new AccountUpdateTransaction()
    .setAccountId(myAccountId)
    .setStakedAccountId("0.0.0")
    .execute(client);
```
3. Submit a transaction (any type) to the network

### Hedera network

mainnet, testnet, previewnet

### Version

latest

### Operating system

None

**Comment by andy-uphold** [comment]
So this effectively kills an account right now.

I have seen it on testnet, but will it have the same effect on mainnet?

If so, is there a workaround? A way to unstake that does work?

**Comment by andy-uphold** [comment]
I'm not expecting a fast fix here, but PLEASE check this for mainnet and update this issue with a comment.

This is blocking me from being able to unstake on mainnet as I do not know if this is safe.

If this affects mainnet then it will make an account unusable if anyone unstakes in this way, so users URGENTLY need to know if:

1) It does affect mainnet
2) If so, if there is a different way to unstake that does NOT kill an account

I am not asking for an immediate fix, but PLEASE answer these concerns.

**Comment by netopyr** [comment]
Fixed in 0.51
